pkg/controller/csi_handler.go

```diff
@@ -458,9 +482,18 @@ func (h *csiHandler) getNodeID(driver string, nodeName string) (string, error) {
 
 	// Check Node annotation.
 	node, err := h.nodeLister.Get(nodeName)
-	if err != nil {
+	if err == nil {
+		return GetNodeIDFromNode(driver, node)
+	}
+
+	// Check VolumeAttachment annotation as the last resort
+	if va == nil {
 		return "", err
```

Comment:
this error will not be 100% accurate, it is just returning the nodeLister error, but in reality it should also contain info about `va` being nil.

Reply:
`va` is nil only only caller wants so. For Attach, I don't want to use annotation in the VA, because that's odd. For Detach, I want to use the annotation. It's not an error if it's nil.

Reply:
I added some comments around.